crates/viewer/re_view_spatial/src/transform_cache.rs

```diff
@@ -850,24 +924,84 @@ fn query_and_resolve_instance_poses_at_entity(
     entity_path: &EntityPath,
     entity_db: &EntityDb,
     query: &LatestAtQuery,
+) -> PoseTransformArchetypeMap {
+    let instance_poses_archetype = query_and_resolve_instance_poses_for_archetype_name(
+        entity_path,
+        entity_db,
+        query,
+        archetypes::InstancePoses3D::name(),
+        &archetypes::InstancePoses3D::descriptor_translations(),
+    );
+
+    // Some archetypes support their own instance poses.
+    // TODO(andreas): can we quickly determine whether this is necessary for any given archetype?
+    let mut per_archetype = IntMap::default();
+    for (archetype_name, descriptor_translations) in
```

Comment:
let's try to improve this a bit by doing a single large query and then operate on the query result instead!